app/javascript/pages/account/components/NameFieldset.tsx

```diff
@@ -13,23 +16,49 @@ interface NameFieldsetProps {
 }
 
 export const handleNameServerErrors =
-  (setError: (name: string, error: ErrorOption) => void) => () => {
-    setError("firstName", { message: "error:name:genericServer", shouldFocus: true })
-  }
-
-export const nameErrorsMap = (errorKey: string, abbreviated: boolean) => {
-  if (errorKey) {
-    switch (errorKey) {
-      case "error:firstName":
-        return t("error.account.firstName")
-      case "error:lastName":
-        return t("error.account.lastName")
-      default:
-        return abbreviated
-          ? t("error.account.genericServerError.abbreviated")
-          : t("error.account.genericServerError")
+  (setError: (name: string, error: ErrorOption) => void) =>
+  (
+    error?: AxiosError<{
+      errors: { full_messages: string[]; lastName: string[]; firstName: string[] }
+    }>
+  ) => {
+    if (
+      error?.response?.status === 422 &&
+      error?.response?.data?.errors?.full_messages.length > 0
+    ) {
+      if (error?.response?.data?.errors?.firstName) {
+        // In the case that the name contains invalid characters (http, www, .), we will show the generic server error
+        if (error.response.data?.errors?.full_messages.includes("Firstname is empty")) {
```

Comment:
I feel like the complexity is coming from a lot of duplicated logic, so the logic can instead be reused for each error type

As just an example, could instead be something like: 
```
export const handleNameServerErrors = (
  key: string,
  error: AxiosError<{
    errors: {
      email: string[]
      password: string[]
      DOB: string[]
      full_messages: string[]
      firstName: string[]
      lastName: string[]
    }
  }>
) => {
  if (error?.response?.status === 422 && error?.response?.data?.errors?.full_messages.length > 0) {
    if (error?.response?.data?.errors?.[key]) {
      // In the case that the name contains invalid characters (http, www, .), we will show the generic server error
      return error.response.data?.errors?.full_messages.includes("is empty")
        ? { message: `name:${key}`, shouldFocus: true }
        : { message: "name:server:generic", shouldFocus: true }
    }

    return { message: "name:server:generic", shouldFocus: true }
  }
}
```

and then in `account-settings.tsx`
```
    if (error.response?.data?.errors?.firstName) {
      setError("firstName", handleNameServerErrors("firstName", error))
    } else if (error.response?.data?.errors?.lastName) {
      setError("lastName", handleNameServerErrors("lastName", error))
    }
```